src/pages/filter/ui/query-builder/groups/chart/pie-chart/pie-chart.tsx

```diff
@@ -51,22 +55,30 @@ export const PieChart = ({ data }: IPieChartProps): ReactElement | null => {
           const optionPercentage = ((value / totalCountsOnChart) * 100).toFixed(
             2,
           )
+          const varaintsQuantityContent = t('filter.chart.variants', {
+            value: formatNumber(value),
+          })
+
+          const isLongInfo = (name + varaintsQuantityContent).length > 30
+
+          if (
+            !isLongInfo &&
+            labelsInCollapsedMode !== extendedLabelsInCollapseMode
+          ) {
+            setLabelsInCollapsedMode(extendedLabelsInCollapseMode)
```

Comment:
It's a bad practice to call `setState` in render flow.

But you still can solve this issue with css only. Try to use flex-box layout in column mode, max-height, flex-wrap and overflow: hidden

Reply:
Fixed it